### PR TITLE
PYTHON-3726 Migrate off of Ubuntu in EG Builds

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: .evergreen/config.yml [hunk 1]
```diff
@@ -2272,10 +2273,11 @@ tasks:
             set -o errexit
             ${PREPARE_SHELL}
             cd src
-            PYTHON_BINARY=
+            export SET_XTRACE_ON=1
```
[the remark this answers is not in the document]
Comment (blink1073) [reply .evergreen/config.yml]: That was for debugging, removed

File: .evergreen/config.yml [hunk 2]
```diff
@@ -2955,21 +2959,6 @@ buildvariants:
   display_name: "${green-framework} ${python-version} ${platform} ${auth-ssl}"
   tasks: *all-server-versions
 
-- matrix_name: "tests-python-version-green-framework-ubuntu20"
```
[the remark this answers is not in the document]
Comment (ShaneHarvey) [reply .evergreen/config.yml:2958]: Aren't those the python version we're excluding:
```
  exclude_spec:
   # Don't test green frameworks on these Python versions.
   - platform: rhel8
     python-version: ["pypy3.7", "pypy3.8", "3.11"]
     green-framework: "*"
     auth-ssl: "*"
```

[the remark this answers is not in the document]
Comment (blink1073) [reply .evergreen/config.yml:2958]: Oof, whoops, fixed

File: .evergreen/config.yml [hunk 3]
```diff
@@ -2322,17 +2323,19 @@ axes:
...
+        display_name: "RHEL 7.x"
         run_on: rhel76-small
         batchtime: 10080  # 7 days
         variables:
+          python3_binary: "/opt/python/3.8/bin/python3"
           libmongocrypt_url: https://s3.amazonaws.com/mciuploads/libmongocrypt/rhel-70-64-bit/master/latest/libmongocrypt.tar.gz
-      - id: rhel84
-        display_name: "RHEL 8.4"
+      - id: rhel8
+        display_name: "RHEL 8.x"
```
Comment (ShaneHarvey) [inline .evergreen/config.yml:2334]: Do we need to change these? Changing the matrix_name/display_name makes evergreen "loose" the task history so I think we should avoid doing it unnecessarily. For example it makes it harder to see when a task started failing or when it last succeeded.

Comment (blink1073) [reply .evergreen/config.yml:2334]: The idea is that we won't need to change again when there are newer RHEL minor versions.  For example, RHEL 7.9 is the only version that will continue to get updates, so we have to migrate.  If we make this change once, we don't lose task history as we have to increment later.

**Review by ShaneHarvey — APPROVED** [review]

File: .evergreen/config.yml (hunk 3 above)
Comment (ShaneHarvey) [reply .evergreen/config.yml:2334]: SGTM

**Pull request merged**